src/main/java/com/xpn/xwiki/plugin/lucene/searcherProvider/SearcherProvider.java

```diff
@@ -38,82 +41,72 @@
    * List of Lucene indexes used for searching. By default there is only one such index
    * for all the wiki. One searches is created for each entry in {@link #indexDirs}.
    */
-  private Searcher[] searchers;
+  private final Searcher[] backedSearchers;
+
+  private volatile boolean markToClose;
 
-  private boolean markToClose;
+  private volatile boolean isClosed;
 
   private Set<Thread> connectedThreads;
 
-  private HashMap<Thread, Set<SearchResults>> connectedSearchResultsMap;
+  private ConcurrentMap<Thread, Set<SearchResults>> connectedSearchResultsMap;
```

Comment:
You are using Thread as key, but what does Thread.hashCode() do? Consider using Thread.getId() as key instead.